rust/private/rust.bzl

```diff
@@ -15,15 +15,6 @@
 load(":private/rustc.bzl", "CrateInfo", "rustc_compile_action")
 load(":private/utils.bzl", "find_toolchain", "relative_path")
 
-def _find_crate_root_src(srcs, file_names = ["lib.rs"]):
-    """Finds the source file for the crate root."""
-    if len(srcs) == 1:
-        return srcs[0]
-    for src in srcs:
-        if src.basename in file_names:
```

Comment:
Needed to check for lib.rs/main.rs before $crate.rs, so had to refactor a little bit.